Tasks/Maven/maven.ps1

```diff
@@ -28,14 +32,37 @@ import-module "Microsoft.TeamFoundation.DistributedTask.Task.Internal"
 import-module "Microsoft.TeamFoundation.DistributedTask.Task.Common"
 import-module "Microsoft.TeamFoundation.DistributedTask.Task.TestResults"
 
-if($jdkVersion -and $jdkVersion -ne "default")
+# If JAVA_HOME is being set by choosing a JDK version find the path to that specified version else use the path given by the user
+$jdkPath = $null
+if($javaHomeSelection -eq 'JDKVersion')
 {
-    $jdkPath = Get-JavaDevelopmentKitPath -Version $jdkVersion -Arch $jdkArchitecture
-    if (!$jdkPath) 
+    Write-Verbose "Using JDK version to find and set JAVA_HOME"
+    # If the JDK version is not the deafult set the jdkPath to the new JDK version selected
+    if($jdkVersion -and $jdkVersion -ne "default")
```

Comment:
This is probably OK, but every powershell book I've read recommends "just put parentheses". I haven't come across any reference to the order of operations that the conditional operators are evaluated, so I just always put parentheses, e.g. if ($jdkVersion -and ($jdkVersion -ne "default"))

Either way, it's your call. Just point it out.